synthesizer/src/vm/verify.rs

```diff
@@ -188,50 +190,9 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
                 );
                 // Verify the signature corresponds to the transaction ID.
                 ensure!(owner.verify(*deployment_id), "Invalid owner signature for deployment transaction '{id}'");
-                // If the `CONSENSUS_VERSION` is less than `V8`, ensure that
-                //   - the deployment edition is zero.
-                // If the `CONSENSUS_VERSION` is less than `V9` ensure that
-                //   - the deployment edition is zero or one.
-                //   - the program checksum is **not** present in the deployment,
-                //   - the program owner is **not** present in the deployment
-                //   - the program does not use constructors, `Operand::Checksum`, `Operand::Edition`, or `Operand::ProgramOwner`
-                // If the `CONSENSUS_VERSION` is greater than or equal to `V9`, then verify that:
-                //   - the program checksum is present in the deployment
-                //   - the program owner is present in the deployment
-                if consensus_version < ConsensusVersion::V8 {
-                    ensure!(
-                        deployment.edition().is_zero(),
-                        "Invalid deployment transaction '{id}' - edition should be zero before `ConsensusVersion::V8`",
-                    );
-                }
-                if consensus_version < ConsensusVersion::V9 {
-                    ensure!(
-                        deployment.edition() <= 1,
-                        "Invalid deployment transaction '{id}' - edition should be zero or one for before `ConsensusVersion::V9`"
-                    );
-                    ensure!(
-                        deployment.program_checksum().is_none(),
-                        "Invalid deployment transaction '{id}' - should not contain program checksum"
-                    );
-                    ensure!(
-                        deployment.program_owner().is_none(),
-                        "Invalid deployment transaction '{id}' - should not contain program owner"
-                    );
-                    ensure!(
-                        !deployment.program().contains_v9_syntax(),
-                        "Invalid deployment transaction '{id}' - program uses syntax that is not allowed before `ConsensusVersion::V9`"
-                    );
-                }
-                if consensus_version >= ConsensusVersion::V9 {
-                    ensure!(
-                        deployment.program_checksum().is_some(),
-                        "Invalid deployment transaction '{id}' - missing program checksum"
-                    );
-                    ensure!(
-                        deployment.program_owner().is_some(),
-                        "Invalid deployment transaction '{id}' - missing program owner"
-                    );
-                }
+
+                let stack = self.process.read().get_stack(deployment.program_id())?;
```

Comment:
Is the stack guaranteed to exist at this point?

Reply:
Ooff good catch. No it's not. And we don't even need it, we just read the `program` from it.